lib/linearize.js

```diff
@@ -510,6 +513,28 @@ class BranchSnapshot {
     return this._rebuild(await this.autobase.latest())
   }
 
+  async updateSession (session) {
+    if (this.isRoot || !session._snapshotted) return this.update()
+    // If the session is a snapshot and this LinearizedCore is not the root, migrate the session to a clone of the root.
+    await this.root.update()
+    const rootClone = this.root.clone()
+
+    const head = this._sessions.pop()
+    if (head !== session) {
+      this._sessions[session._sessionIdx] = head
+      head._sessionIdx = session._sessionIdx
+    }
+
+    session._sessionIdx = rootClone._sessions.push(session) - 1
```

Comment:
you don't need `this._sessions = rootClone._sessions` in case this is called again?